packages/overlay/src/overlay-trigger-directive.ts

```diff
@@ -0,0 +1,142 @@
+    private abortController!: AbortController;
+
+    protected defaultOptions: OverlayTriggerOptions = {
+        triggerInteraction: 'hover',
+        overlayOptions: {
+            placement: 'top-start',
+            type: 'auto',
+            offset: 0,
+        },
+    };
+    protected options: OverlayOptions = {
+        ...this.defaultOptions.overlayOptions,
+    };
+
+    render(
+        _template: () => TemplateResult,
+        _options?: Partial<OverlayTriggerOptions>
+    ): unknown {
+        // render function here just defines the interface to the update call later
+        // we don't have anything to render since this is tintended to be an ElementPart directive
+        // so will be used on an element and is not itself rendered
+        return nothing;
+    }
+
+    override update(
+        part: ElementPart,
+        [template, options]: Parameters<this['render']>
+    ): void {
+        this.options = {
+            ...this.defaultOptions.overlayOptions,
+            ...options?.overlayOptions,
+        };
+        this.template = template;
+        let newTarget = false;
+        if (this.target !== part.element) {
+            this.target = part.element as HTMLElement;
+            newTarget = true;
+        }
+        if (
+            newTarget ||
+            this.strategy?.type !== options?.triggerInteraction ||
+            this.defaultOptions.triggerInteraction
+        ) {
+            this.strategy?.abort();
+            this.strategy = new strategies[
+                options?.triggerInteraction ||
+                    this.defaultOptions.triggerInteraction
+            ](this.overlay, this.target, true);
+        }
+        this.init();
+
+        if (window.__swc.DEBUG) {
+            window.__swc.warn(
+                undefined,
+                `⚠️  WARNING ⚠️ : The Overlay Trigger Directive is experimental and there is no guarantees behind its usage in an application!! Its API and presence within the library could be changed at anytime. See "sp-overlay" or "Overlay.open()" for a stable API for overlaying content on your application.`,
+                'https://opensource.adobe.com/spectrum-web-components/components/overlay',
+                {
+                    level: 'high',
+                    type: 'api',
+                }
+            );
+        }
+    }
+
+    handleSlottableRequest(event: SlottableRequestEvent): void {
+        if (event.target !== event.currentTarget) return;
+
+        const willRemoveSlottable = event.data === removeSlottableRequest;
+
+        render(willRemoveSlottable ? undefined : this.template(), this.overlay);
+        if (willRemoveSlottable) {
+            this.overlay.remove();
+        } else {
+            Overlay.applyOptions(this.overlay, {
+                ...this.options,
+                trigger: this.target,
+            });
+            this.target.insertAdjacentElement('afterend', this.overlay);
```

Comment:
Should we make the insertion point configurable or is after the target always ok?

Reply:
Added.